### PR TITLE
Handle string IDs in the article content voter

### LINKED ISSUE #7508: Error 500: ArticleContentVoter::getPageId(): Return value must be of type ?int, string returned
### Affected version(s)

5.3.13

### The lastest update **5.3.12 -> 5.3.13** completely blocked the ability to edit "Content/Articles" articles.

When I try to go there through the menu I get a **500 error** with the message:

> "Contao\CoreBundle\Security\Voter\DataContainer\ArticleContentVoter::getPageId(): Return value must be of type ?int, string returned"

Here is a fragment of the log entry

> [2024-09-04T08:08:11.826034+02:00] request.CRITICAL: Uncaught PHP Exception TypeError: "Contao\CoreBundle\Security\Voter\DataContainer\ArticleContentVoter::getPageId(): Return value must be of type ?int, string returned" at ArticleContentVoter.php line 69 {"exception":"[object] (TypeError(code: 0): Contao\\CoreBundle\\Security\\Voter\\DataContainer\\ArticleContentVoter::getPageId(): Return value must be of type ?int, string returned at /home/.../public_html/vendor/contao/core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php:69)"} {"request_uri":"https://www...pl/admin?do=article&ref=mgLbnYQR","request_method":"GET"}

The problem occurred on all pages that I updated to 5.13.13

The downgrade **5.13.13 -> 5.13.12** helps to solve the problem.

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/7508

### CODE REVIEW

File: core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php [hunk 1]
```diff
@@ -62,10 +62,11 @@ protected function hasAccessToRecord(TokenInterface $token, string $table, int $
 
     private function getPageId(int $articleId): int|null
     {
-        if (!isset($this->pageIds[$articleId])) {
-            $this->pageIds[$articleId] = $this->connection->fetchOne('SELECT pid FROM tl_article WHERE id=?', [$articleId]);
+        if (!array_key_exists($articleId, $this->pageIds)) {
+            $pid = $this->connection->fetchOne('SELECT pid FROM tl_article WHERE id=?', [$articleId]);
+            $this->pageIds[$articleId] = $pid ? (int) $pid : null;
```
Comment (m-vo) [inline core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php]: If we're at it, should we also harden the check to allow for `0` as a value?

```suggestion
            $this->pageIds[$articleId] = false !== $pid ? (int) $pid : null;
```

Comment (aschempp) [reply core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php]: `0` wouldn't be a valid return value for that method, it wouldn't be a page ID

Comment (m-vo) [reply core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php]: If it is fetched from the db it IMHO is. You can modify an INT PK to be 0 on db side without problems and it has no special meaning in Contao, does it?

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**